packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.ts

```diff
@@ -85,6 +90,23 @@ export abstract class XYChartComponent extends TimeseriesChartComponent
         }
     }
 
+    public onPrimaryDescClick(legendSeries: IChartAssistSeries<IAccessors>) {
+        if (!this.seriesInteractive) {
+            this.chartAssist.toggleSeries(legendSeries.id, this.chartAssist.isSeriesHidden(legendSeries.id));
```

Comment:
This shouldn't be necessary if you change it to conditionally `stopPropagation` only if `this.seriesInteractive`. Currently, `stopPropagation` is invoked every time the legend is clicked.